cedar-policy-core/src/ast/id.rs

```diff
@@ -79,6 +90,48 @@ impl FromNormalizedStr for Id {
     }
 }
 
+/// An `Id` that is not equal to `__cedar`, as specified by RFC 52
+#[derive(Serialize, Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord)]
+pub struct UnreservedId(pub(crate) Id);
```

Comment:
I'm a little concerned about having three identifier types `Id`, `AnyId`, and `UnreservedId` without more guidance for when each should be used.

Reply:
I agree. This PR also adds `UnreservedName` in addition to the existing `Name`.

Reply:
I tried to replace `Name` with `UnservedName` but realized that we may have future use cases for the former.